File: examples/Scala/wuproxy.scala
/*
 *
 *  Weather proxy device in Scala
 *
 * @author Vadim Shalts
 * @email vshalts@gmail.com
*/

import org.zeromq.ZMQ

object wuproxy {

  def main(args: Array[String]) {

    //  Prepare our context and sockets
    var context = ZMQ.context(1)

    //  This is where the weather server sits
    var frontend = context.socket(ZMQ.SUB)
    frontend.connect("tcp://192.168.55.210:5556")

    //  This is our public endpoint for subscribers
    var backend = context.socket(ZMQ.PUB)
    backend.bind("tcp://10.1.1.0:8100")

    //  Subscribe on everything
    frontend.subscribe("".getBytes)

    //  Shunt messages out to our own subscribers
    while (!Thread.currentThread.isInterrupted) {
      var more = false
      do {
        var message = frontend.recv(0)
        more = frontend.hasReceiveMore
        backend.send(message, if (more) ZMQ.SNDMORE else 0)
      } while(more)
    }

    frontend.close()
    backend.close()
    context.term()
  }
}

File: examples/Go/wuproxy.go
// Weather proxy device
//
// Author:  Brendan Mc.
// Requires: http://github.com/alecthomas/gozmq

package main

import (
    zmq "github.com/alecthomas/gozmq"
)

func main() {
    context, _ := zmq.NewContext()
	defer context.Close()
    
    // This is where the weather server sits
    frontend, _ := context.NewSocket(zmq.SUB)
    defer frontend.Close()
    frontend.Connect("tcp://localhost:5556")
    
    // This is our public endpoint for subscribers
    backend, _ := context.NewSocket(zmq.PUB)
    defer backend.Close()
    backend.Bind("tcp://*:8100")
    
    // Subscribe on everything
    frontend.SetSockOptString(zmq.SUBSCRIBE, "")
    
    // Shunt messages out to our own subscribers
    for {
        message, _ := frontend.Recv(0)
        backend.Send(message, 0)
    }
}
